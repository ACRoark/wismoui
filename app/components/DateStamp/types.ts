export type DateStampFormat = 'auto' | 'long' | 'short';
