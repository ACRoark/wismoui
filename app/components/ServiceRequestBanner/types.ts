export interface IBannerNotification {
  hasError: boolean;
  message: IMessage;
}

export interface IMessage {
  id: string;
  values: string[] | null;
}
