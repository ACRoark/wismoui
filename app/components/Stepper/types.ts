import { ReactElement } from 'react';
import { Description } from 'types';

export interface IStepDescription {
  description: Description;
  key: string;
  status: StepperStatus;
  title: ReactElement | string;
}

export enum StepperStatus {
  Active = 'active',
  Completed = 'completed',
  Error = 'error',
  Pending = 'pending',
}
