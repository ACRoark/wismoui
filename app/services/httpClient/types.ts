export interface IHttpError extends Error {
  code: number;
  reason: string;
}

export interface IHttpResponse {
  // tslint:disable-next-line: no-any
  data: any;
  status: number;
  statusText: string;
}

export interface IHttpClient {
  cancel: () => void;
  // tslint:disable-next-line: no-any
  get(url: string, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  post(url: string, data: any, config?: any): Promise<IHttpResponse>;
  // tslint:disable-next-line: no-any
  wasCancelled: (error: any) => boolean;
}
