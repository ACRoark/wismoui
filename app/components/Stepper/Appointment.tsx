import React from 'react';
import { FormattedMessage } from 'react-intl';

import { SlotType } from 'types';

import getAppointmentType from './getAppointmentType';
import getTime from './getTime';
import messages from './messages';

interface IAppointmentProps {
  product: string;
  slotType: SlotType;
}

const Appointment: React.FC<IAppointmentProps> = (props: IAppointmentProps): React.ReactElement => {
  const { product, slotType } = props;

  const time = getTime(slotType);
  const serviceType = getAppointmentType(product);

  return (
    <>
      {product && slotType && slotType !== 'AN' && (
        <div>
          <FormattedMessage {...messages[serviceType]} values={{ window: time }} />
        </div>
      )}
    </>
  );
};

export default Appointment;
