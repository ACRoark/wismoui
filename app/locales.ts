export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
