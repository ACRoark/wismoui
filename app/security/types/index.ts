import IAuthenticationState from 'types/IAuthenticationState';

import IAuthenticationResult from './IAuthenticationResult';

export { IAuthenticationResult };

export default IAuthenticationState;
