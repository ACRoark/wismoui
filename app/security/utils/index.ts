import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
