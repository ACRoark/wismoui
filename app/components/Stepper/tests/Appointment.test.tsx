import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';
import { SlotType, SlotTypes } from 'types';

import Appointment from '../Appointment';

const expectComponentToMatchSnapshot = (language: string, product: string, slotType: SlotType): void => {
  const tree = createSnapshotWithIntl(<Appointment product={product} slotType={slotType}/>, language);

  expect(tree).toMatchSnapshot();
};

const expectNoConsoleErrors = (language: string, product: string, slotType: SlotType): void => {
  const spy = jest.spyOn(global.console, 'error');

  renderWithIntl(<Appointment product={product} slotType={slotType}/>, language);

  expect(spy).not.toHaveBeenCalled();
};

describe('Appointment component', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  SlotTypes.forEach((slotType: SlotType): void => {
    languages.forEach((language: string): void => {
      it(`should not log errors in console when rendered for a residential electric product,
      the appointment time is ${slotType} and the current language is ${language}`, (): void =>
        expectNoConsoleErrors(language, 'EFR_D1', slotType));

      it(`should not log errors in console when rendered for a residential gas product,
      the appointment time is ${slotType} and the current language is ${language}`, (): void =>
        expectNoConsoleErrors(language, 'GCR_A_H', slotType));

      it(`should render properly when given a residential electric product,
      the appointment time is ${slotType} and the current language is ${language}`, (): void =>
        expectComponentToMatchSnapshot(language, 'EFR_D1', slotType));

      it(`should render properly when given a residential gas product,
      the appointment time is ${slotType} and the current language is ${language}`, (): void =>
        expectComponentToMatchSnapshot(language, 'GCR_A_H', slotType));

      it(`should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is ${slotType} and the current language is ${language}`, (): void =>
        expectComponentToMatchSnapshot(language, 'INTERNET', slotType));
    });
  });
});
