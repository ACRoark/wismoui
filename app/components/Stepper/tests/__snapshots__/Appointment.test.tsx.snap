// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is AH and the current language is en 1`] = `
<div>
  Electric appointment 
  4:30 p.m. - 8 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is AH and the current language is es 1`] = `
<div>
  Cita electrica 
  4:30 p.m. - 8 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is AM and the current language is en 1`] = `
<div>
  Electric appointment 
  8 a.m. - 12 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is AM and the current language is es 1`] = `
<div>
  Cita electrica 
  8 a.m. - 12 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is AN and the current language is en 1`] = `null`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is AN and the current language is es 1`] = `null`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is PM and the current language is en 1`] = `
<div>
  Electric appointment 
  12 p.m. - 4:30 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is PM and the current language is es 1`] = `
<div>
  Cita electrica 
  12 p.m. - 4:30 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is null and the current language is en 1`] = `null`;

exports[`Appointment component should render properly when given a residential electric product,
      the appointment time is null and the current language is es 1`] = `null`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is AH and the current language is en 1`] = `
<div>
  Natural Gas appointment 
  4:30 p.m. - 8 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is AH and the current language is es 1`] = `
<div>
  Cita de gas natural 
  4:30 p.m. - 8 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is AM and the current language is en 1`] = `
<div>
  Natural Gas appointment 
  8 a.m. - 12 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is AM and the current language is es 1`] = `
<div>
  Cita de gas natural 
  8 a.m. - 12 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is AN and the current language is en 1`] = `null`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is AN and the current language is es 1`] = `null`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is PM and the current language is en 1`] = `
<div>
  Natural Gas appointment 
  12 p.m. - 4:30 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is PM and the current language is es 1`] = `
<div>
  Cita de gas natural 
  12 p.m. - 4:30 p.m.
</div>
`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is null and the current language is en 1`] = `null`;

exports[`Appointment component should render properly when given a residential gas product,
      the appointment time is null and the current language is es 1`] = `null`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is AH and the current language is en 1`] = `
<div>
  Service appointment 
  4:30 p.m. - 8 p.m.
</div>
`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is AH and the current language is es 1`] = `
<div>
  Cita de servicio 
  4:30 p.m. - 8 p.m.
</div>
`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is AM and the current language is en 1`] = `
<div>
  Service appointment 
  8 a.m. - 12 p.m.
</div>
`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is AM and the current language is es 1`] = `
<div>
  Cita de servicio 
  8 a.m. - 12 p.m.
</div>
`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is AN and the current language is en 1`] = `null`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is AN and the current language is es 1`] = `null`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is PM and the current language is en 1`] = `
<div>
  Service appointment 
  12 p.m. - 4:30 p.m.
</div>
`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is PM and the current language is es 1`] = `
<div>
  Cita de servicio 
  12 p.m. - 4:30 p.m.
</div>
`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is null and the current language is en 1`] = `null`;

exports[`Appointment component should render the default Service appointment when given a product that is neither gas or electric,
      the appointment time is null and the current language is es 1`] = `null`;
